[Bug] --queries-timeout 60.0 invalid type: floating point

### Describe the bug

Running the bridge I get following error with the queries timeout arg:


```
ros2 run zenoh_bridge_ros2dds zenoh_bridge_ros2dds \
        -l tcp/0.0.0.0:7447 \
        --no-multicast-scouting \
        --queries-timeout 60.0
[2023-12-10T13:34:21Z INFO  zenoh_bridge_ros2dds] zenoh-bridge-ros2dds veb66be5 built with rustc 1.70.0 (90c541806 2023-05-31) (built from a source tarball)
[2023-12-10T13:34:21Z INFO  zenoh::net::runtime] Using PID: 6b8adfe3b7ac3d9f50bb8cce3dcdc2c4
[2023-12-10T13:34:21Z INFO  zenoh::net::runtime::orchestrator] Zenoh can be reached at: tcp/192.168.178.32:7447
[2023-12-10T13:34:21Z INFO  zenoh::net::runtime::orchestrator] Zenoh can be reached at: tcp/172.17.0.1:7447
[2023-12-10T13:34:21Z INFO  zenoh::net::runtime::orchestrator] Zenoh can be reached at: tcp/172.22.0.1:7447
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Plugin `ros2dds` configuration error: invalid type: floating point `60`, expected struct QueriesTimeouts at zenoh-plugin-ros2dds/src/lib.rs:130.', zenoh-bridge-ros2dds/src/main.rs:220:66
stack backtrace:
   0:     0x61183eabcef7 - std::backtrace_rs::backtrace::libunwind::trace::hc8c748621e1717dc
                               at /build/rustc-wAuwbs/rustc-1.70.0+dfsg0ubuntu1~bpo2/library/std/src/../../backtrace/src/backtrace/libunwind.rs:93:5
   1:     0x61183eabcef7 - std::backtrace_rs::backtrace::trace_unsynchronized::h1657ffb548bcd1c0
                               at /build/rustc-wAuwbs/rustc-1.70.0+dfsg0ubuntu1~bpo2/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2:     0x61183eabcef7 - std::sys_common::backtrace::_print_fmt::ha3920362e42412b1
                               at /build/rustc-wAuwbs/rustc-1.70.0+dfsg0ubuntu1~bpo2/library/std/src/sys_common/backtrace.rs:65:5
   3:     0x61183eabcef7 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h55a64e3639141d5c
                               at /build/rustc-wAuwbs/rustc-1.70.0+dfsg0ubuntu1~bpo2/library/std/src/sys_common/backtrace.rs:44:22
   4:     0x61183e8c9bdf - core::fmt::write::h8a6836b78d5c6e17
                               at /build/rustc-wAuwbs/rustc-1.70.0+dfsg0ubuntu1~bpo2/library/core/src/fmt/mod.rs:1254:17
   5:     0x61183eaca8f4 - std::io::Write::write_fmt::h74083ba874a8bcaf
                               at /build/rustc-wAuwbs/rustc-1.70.0+dfsg0ubuntu1~bpo2/library/std/src/io/mod.rs:1698:15
   6:     0x61183eabcb9f - std::sys_common::backtrace::_print::he2a2a37b05e7e764
                               at /build/rustc-wAuwbs/rustc-1.70.0+dfsg0ubuntu1~bpo2/library/std/src/sys_common/backtrace.rs:47:5
   7:     0x61183eabcb9f - std::sys_common::backtrace::print::hf1818e0fa9ba5bd4
                               at /build/rustc-wAuwbs/rustc-1.70.0+dfsg0ubuntu1~bpo2/library/std/src/sys_common/backtrace.rs:34:9
   8:     0x61183eac80ee - std::panicking::default_hook::{{closure}}::h36f8b00cf3f70373
                               at /build/rustc-wAuwbs/rustc-1.70.0+dfsg0ubuntu1~bpo2/library/std/src/panicking.rs:269:22
   9:     0x61183eac8c41 - std::panicking::default_hook::h0999e9f322268afa
                               at /build/rustc-wAuwbs/rustc-1.70.0+dfsg0ubuntu1~bpo2/library/std/src/panicking.rs:288:9
  10:     0x61183eac8c41 - std::panicking::rust_panic_with_hook::hf0f1736a09665bd3
                               at /build/rustc-wAuwbs/rustc-1.70.0+dfsg0ubuntu1~bpo2/library/std/src/panicking.rs:691:13
  11:     0x61183eabd304 - std::panicking::begin_panic_handler::{{closure}}::h9c059812511f285d
                               at /build/rustc-wAuwbs/rustc-1.70.0+dfsg0ubuntu1~bpo2/library/std/src/panicking.rs:582:13
  12:     0x61183eabd266 - std::sys_common::backtrace::__rust_end_short_backtrace::haeaaa791abdbfc5f
                               at /build/rustc-wAuwbs/rustc-1.70.0+dfsg0ubuntu1~bpo2/library/std/src/sys_common/backtrace.rs:150:18
  13:     0x61183eac84f1 - rust_begin_unwind
                               at /build/rustc-wAuwbs/rustc-1.70.0+dfsg0ubuntu1~bpo2/library/std/src/panicking.rs:578:5
  14:     0x61183e773fc2 - core::panicking::panic_fmt::h42c1b10c64151f1e
                               at /build/rustc-wAuwbs/rustc-1.70.0+dfsg0ubuntu1~bpo2/library/core/src/panicking.rs:67:14
  15:     0x61183e773e72 - core::result::unwrap_failed::h4b24ec377bba078a
                               at /build/rustc-wAuwbs/rustc-1.70.0+dfsg0ubuntu1~bpo2/library/core/src/result.rs:1687:5
  16:     0x61183e7af5d1 - <async_std::task::builder::SupportTaskLocals<F> as core::future::future::Future>::poll::h31d76bd20aab43e1
  17:     0x61183e8330d4 - zenoh_bridge_ros2dds::main::h0c723d0e67799cff
  18:     0x61183e7f15b3 - std::sys_common::backtrace::__rust_begin_short_backtrace::h1418525716f9748f
  19:     0x61183e8343d3 - main
  20:     0x71ab3632ad90 - __libc_start_call_main
                               at ./csu/../sysdeps/nptl/libc_start_call_main.h:58:16
  21:     0x71ab3632ae40 - __libc_start_main_impl
                               at ./csu/../csu/libc-start.c:392:3
  22:     0x61183e7a85b5 - _start
  23:                0x0 - <unknown>
[ros2run]: Aborted
make: *** [Makefile:54: zenoh] Error 250
```

### To reproduce

Run 

```
ros2 run zenoh_bridge_ros2dds zenoh_bridge_ros2dds \
		-l tcp/0.0.0.0:7447 \
		--no-multicast-scouting \
		--queries-timeout 60.0
```

### System info

- Ubuntu 22.04 in docker image based on osrf/ros:humble-desktop-full
- zenoh-plugin-ros2dds hash: eb66be54a0f5af584134843fe86a2fa02493496c

--- a/DEFAULT_CONFIG.json5
+++ b/DEFAULT_CONFIG.json5
@@ -98,12 +98,15 @@
 
       ////
       //// queries_timeout: Timeouts configuration for various Zenoh queries.
-      ////                  Each field is optional. If not set, A default value of 5.0 seconds applies.
+      ////                  Each field is optional. If not set, the 'default' timeout (5.0 seconds by default) applies to all queries.
       ////                  Each value can be either a float in seconds that will apply as a timeout to all queries,
       ////                  either a list of strings with format "<regex>=<float>" where:
       ////                      - "regex" is a regular expression matching an interface name
       ////                      - "float" is the timeout in seconds
       // queries_timeout: {
+      //   //// default timeout that will apply to all query, except the ones specified below
+      //   //// in 'transient_local_subscribers', 'services' and 'actions'
+      //   default: 5.0,
       //   //// timeouts for TRANSIENT_LOCAL subscriber when querying publishers for historical publications
       //   transient_local_subscribers: 1.0,
       //   //// timeouts for Service clients calling a Service server

--- a/zenoh-bridge-ros2dds/src/main.rs
+++ b/zenoh-bridge-ros2dds/src/main.rs
@@ -118,9 +118,10 @@ r#"--pub-max-frequency=[String]...   'Specifies a maximum frequency of publicati
 Repeat this option to configure several topics expressions with a max frequency.'"#
         ))
         .arg(Arg::from_usage(
-r#"--queries-timeout=[float]... 'A float in seconds (default: 5.0 sec) that will be used as a timeout when the bridge
+r#"--queries-timeout-default=[float]... 'A float in seconds (default: 5.0 sec) that will be used as a timeout when the bridge
 queries any other remote bridge for discovery information and for historical data for TRANSIENT_LOCAL DDS Readers it serves
-(i.e. if the query to the remote bridge exceed the timeout, some historical samples might be not routed to the Readers, but the route will not be blocked forever)."#
+(i.e. if the query to the remote bridge exceed the timeout, some historical samples might be not routed to the Readers, but the route will not be blocked forever).
+This value overwrites the value possibly set in configuration file under 'plugins/ros2dds/queries_timeout/default' key."#
         ))
         .arg(Arg::from_usage(
 r#"--watchdog=[PERIOD]   'Experimental!! Run a watchdog thread that monitors the bridge's async executor and reports as error log any stalled status during the specified period (default: 1.0 second)'"#
@@ -180,7 +181,7 @@ r#"--watchdog=[PERIOD]   'Experimental!! Run a watchdog thread that monitors the
         insert_json5!(config, args, "plugins/ros2dds/shm_enabled", if "dds-enable-shm");
     }
     insert_json5!(config, args, "plugins/ros2dds/pub_max_frequencies", for "pub-max-frequency", .collect::<Vec<_>>());
-    insert_json5!(config, args, "plugins/ros2dds/queries_timeout", if "queries-timeout", .parse::<f64>().unwrap());
+    insert_json5!(config, args, "plugins/ros2dds/queries_timeout/default", if "queries-timeout-default", .parse::<f32>().unwrap());
 
     let watchdog_period = if args.is_present("watchdog") {
         args.value_of("watchdog").map(|s| s.parse::<f32>().unwrap())

--- a/zenoh-plugin-ros2dds/src/config.rs
+++ b/zenoh-plugin-ros2dds/src/config.rs
@@ -24,10 +24,7 @@ pub const DEFAULT_DOMAIN: u32 = 0;
 pub const DEFAULT_RELIABLE_ROUTES_BLOCKING: bool = true;
 pub const DEFAULT_TRANSIENT_LOCAL_CACHE_MULTIPLIER: usize = 10;
 pub const DEFAULT_DDS_LOCALHOST_ONLY: bool = false;
-
-lazy_static::lazy_static!(
-    pub static ref DEFAULT_QUERIES_TIMEOUT: Duration = Duration::from_secs_f32(5.0);
-);
+pub const DEFAULT_QUERIES_TIMEOUT: f32 = 5.0;
 
 #[derive(Deserialize, Debug, Serialize)]
 #[serde(deny_unknown_fields)]
@@ -81,8 +78,9 @@ impl Config {
                     return Duration::from_secs_f32(*secs);
                 }
             }
+            return Duration::from_secs_f32(qt.default);
         }
-        *DEFAULT_QUERIES_TIMEOUT
+        Duration::from_secs_f32(DEFAULT_QUERIES_TIMEOUT)
     }
 
     pub fn get_queries_timeout_service(&self, ros2_name: &str) -> Duration {
@@ -92,13 +90,16 @@ impl Config {
                     return Duration::from_secs_f32(*secs);
                 }
             }
+            return Duration::from_secs_f32(qt.default);
         }
-        *DEFAULT_QUERIES_TIMEOUT
+        Duration::from_secs_f32(DEFAULT_QUERIES_TIMEOUT)
     }
 
     pub fn get_queries_timeout_action_send_goal(&self, ros2_name: &str) -> Duration {
         if let Some(QueriesTimeouts {
-            actions: Some(at), ..
+            default,
+            actions: Some(at),
+            ..
         }) = &self.queries_timeout
         {
             for (re, secs) in &at.send_goal {
@@ -106,13 +107,16 @@ impl Config {
                     return Duration::from_secs_f32(*secs);
                 }
             }
+            return Duration::from_secs_f32(*default);
         }
-        *DEFAULT_QUERIES_TIMEOUT
+        Duration::from_secs_f32(DEFAULT_QUERIES_TIMEOUT)
     }
 
     pub fn get_queries_timeout_action_cancel_goal(&self, ros2_name: &str) -> Duration {
         if let Some(QueriesTimeouts {
-            actions: Some(at), ..
+            default,
+            actions: Some(at),
+            ..
         }) = &self.queries_timeout
         {
             for (re, secs) in &at.cancel_goal {
@@ -120,13 +124,16 @@ impl Config {
                     return Duration::from_secs_f32(*secs);
                 }
             }
+            return Duration::from_secs_f32(*default);
         }
-        *DEFAULT_QUERIES_TIMEOUT
+        Duration::from_secs_f32(DEFAULT_QUERIES_TIMEOUT)
     }
 
     pub fn get_queries_timeout_action_get_result(&self, ros2_name: &str) -> Duration {
         if let Some(QueriesTimeouts {
-            actions: Some(at), ..
+            default,
+            actions: Some(at),
+            ..
         }) = &self.queries_timeout
         {
             for (re, secs) in &at.get_result {
@@ -134,14 +141,17 @@ impl Config {
                     return Duration::from_secs_f32(*secs);
                 }
             }
+            return Duration::from_secs_f32(*default);
         }
-        *DEFAULT_QUERIES_TIMEOUT
+        Duration::from_secs_f32(DEFAULT_QUERIES_TIMEOUT)
     }
 }
 
 #[derive(Deserialize, Debug, Serialize)]
 #[serde(deny_unknown_fields)]
 pub struct QueriesTimeouts {
+    #[serde(default = "default_queries_timeout")]
+    default: f32,
     #[serde(
         default,
         deserialize_with = "deserialize_vec_regex_f32",
@@ -347,6 +357,10 @@ fn default_domain() -> u32 {
     } else {
         DEFAULT_DOMAIN
     }
+}
+
+fn default_queries_timeout() -> f32 {
+    DEFAULT_QUERIES_TIMEOUT
 }
 
 fn deserialize_path<'de, D>(deserializer: D) -> Result<Option<Vec<String>>, D::Error>

